### DISCUSSION TITLE
Four-Vectors: How to Use Upper and Lower Indices?

### DISCUSSION THREAD

**Discussion by ghost** [FeynCalc/feyncalc#109] [Q&A]
When I type into Mathematica `FVD[p + r, Lower[mu]]`, I get the following output (before the `mu`, I have three vertical dots, so it's the Greek letter): 

![image-mathematica](https://user-images.githubusercontent.com/63545056/113449352-f61c7080-93fd-11eb-8b61-a2997858aba3.png)

Sure, for contraction, it might not matter whether the index is lower or upper - the way it is implemented, but apart from, it *can* matter.. My question is how I can get the index in a lower position of the four-vector.

**Comment by vsht** [comment]
Sec 4.2 in https://arxiv.org/pdf/2006.15451.pdf describes how you can 
distinguish between the two without having explicit upper and lower indices.
If that doesn't suite you, I'd be interested to understand your usage case.

But apart from that, there are no built-in mechanisms to distinguish between
covariant and contravariant indices. The `Upper` and `Lower` symbols 
available in some older versions did not really work except for few very simple
cases and were therefore removed in 9.3.
